### ISSUE TITLE
Review Accessibility compliance

### ISSUE THREAD

**Issue by maxxxxxdlp** [specify/specify7#3380]
On top of the things in https://github.com/specify/specify7/projects/12

- Make sure all dialogs with fields have `<form>` and submit button - this way users can use Enter key to submit the form (which can also be made to close the dialog)
- Try out using screen reader. Apple's VoiceOver might be a good example
- Evaluate usages of role and aria-label. Remove extra usages
- Consider HTML tags: noscript, address, article, aside, section
- Use `<output>` HTML tax for search dialog results, query results, and other things (it adds aria-live, so test screen reader when queryResultsTable fetched more records)
- Evaluate usages of aria-live and aria-atomic (and add missing)
- Review all aria roles: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques
- Learn lessons from https://react-spectrum.adobe.com/blog/building-a-combobox.html
